src/coreclr/jit/gentree.cpp

```diff
@@ -23270,6 +23273,7 @@ GenTree* Compiler::gtNewSimdGetElementNode(
     var_types      simdBaseType = JitType2PreciseVarType(simdBaseJitType);
 
     assert(varTypeIsArithmetic(simdBaseType));
+    assert(op2->TypeIs(TYP_I_IMPL));
```

Comment:
Just noting here, for explanation....

The reason that `gtNewSimdGetElementNode` took `TYP_INT` is because the underlying instruction takes `TYP_BYTE`.

The only time we actually consume it as `TYP_I_IMPL` is if it is non-constant and we must fallback to spilling the vector to memory and using the index dynamically.